oracle.go

```diff
@@ -280,6 +280,72 @@ func (o *Oracle) RotateRootCredentials(ctx context.Context, statements []string)
 	return o.RawConfig, nil
 }
 
+func (o *Oracle) SetCredentials(ctx context.Context, statements dbplugin.Statements, staticUser dbplugin.StaticUserConfig) (username, password string, err error) {
+	rotateStatements := statements.Rotation
+	if len(rotateStatements) == 0 {
+		rotateStatements = []string{`ALTER USER "{{username}}" IDENTIFIED BY "{{password}}"`}
+	}
+
+	username = staticUser.Username
+	password = staticUser.Password
+	if username == "" || password == "" {
+		return "", "", errors.New("must provide both username and password")
+	}
+
+	variables := map[string]string{
+		"username": strings.ToUpper(username),
+		"password": password,
+	}
+
+	queries := splitQueries(rotateStatements)
+	if len(queries) == 0 { // Extra check to protect against future changes
+		return "", "", errors.New("no rotation queries found")
+	}
+
+	// Lock the SQL connection
+	o.Lock()
+	defer o.Unlock()
+
+	db, err := o.getConnection(ctx)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to get database connection: %w", err)
+	}
+
+	tx, err := db.BeginTx(ctx, nil)
+	if err != nil {
+		return "", "", fmt.Errorf("unable to create database transaction: %w", err)
+	}
+	defer tx.Rollback()
+
+	for _, rawQuery := range queries {
+		err := dbtxn.ExecuteTxQuery(ctx, tx, variables, rawQuery)
+		if err != nil {
+			return "", "", fmt.Errorf("unable to execute rotation query: %w", err)
+		}
+	}
+
+	err = tx.Commit()
```

Comment:
Consider making this into a short `if`.

Reply:
I'm personally not a fan of short ifs. I feel that the cons outweigh the pros.

Reply:
What are the cons? By this, do you mean the statement:
```
if err := foo(); err != nil {
  return err
}
```
If so, I'm actually a huge fan of those because they discourage err reduce (which can lead to races if errs are used in and out of a goroutine), and because they keep the err's scope to a minimum. Lots of languages, in principle, encourage minimizing variable scope.

I'm actually very interested to hear why you're not a fan, I imagine you've got an interesting take on it. I hope that doesn't come across as terse, it's not intended that way, I am sincerely curious!

Reply:
I don't find it very valuable because errors are reused in many other circumstances, even inside the same function. This makes the micro-optimization generally useless in my opinion. For instance:
```go
foo, err := doFoo()
if err != nil {
  return err
}

bar, err := doErr()
if err != nil {
  return err
}

// Doing this here doesn't add much protection because err is already being reused in
// other calls which aren't protected.
if err := doBaz(); err != nil {
  return err
}
```

Additionally, there are times where it actually increases the complexity of your code. Consider these two functionally identical snippets:
```go
if foo, err := newFoo(); err != nil {
  return err
} else {
  foo.Foo()
}
```
```go
foo, err := newFoo()
if err != nil {
  return err
}

foo.Foo()
```
Using the inline if created another layer of indentation. This hurts readability and adds unnecessary complexity. This problem exists in the plugin currently: https://github.com/hashicorp/vault-plugin-database-oracle/blob/ac66c8797d8627e6eab75086419c65b1fd5ceb10/oracle.go#L289-L311

Generally speaking you want the "normal" execution of your code as far to the left as possible so it's easy to follow. The exit conditions (usually returning an error) are generally indented from the "normal" case. In the example above, the normal execution was no longer towards the left. It was moved right, hurting readability.